---
title: "Genomics Research Centre Diagnostics Clinic Genotyping Analysis Report"
date: '`r format(Sys.time(), "%d %B, %Y")`'
output: 
  word_document:
    reference_docx: 'template/GRC_WES_report_template_landscape.docx'
    fig_caption: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
require(magrittr)
```

```{r pander_options, echo=FALSE}
# set pander table-layout options
library(pander)
panderOptions('table.alignment.default', function(df)
    ifelse(sapply(df, is.numeric), 'right', 'left'))
panderOptions('table.split.table', Inf)
panderOptions('big.mark', ",")
panderOptions('keep.trailing.zeros', TRUE)
```

```{r sampleID, echo=FALSE, message=FALSE}
# enter sample ID
sampleID <- 'DG515'

# get software and database versions
versionList <- list.files('../vcf', pattern = '_versions.txt', full.names = T)
versionFile <- versionList[grep(sampleID, versionList)]
# load version info
versions <- read.csv(versionFile, head = F, as.is = T)
colnames(versions) <- 'Parameters'
# get run ID
# ...extraction script here...
runID <- '024'  # placeholder
# genome build
genomeBuild <- 'hg19'
```

**Patient:** `r paste(sampleID)`

**RUNID:** `r paste(runID)` **{is this correct?}**

**Analysis Performed:**
Alignment against `r genomeBuild` -> annotation pipeline (dbSNP, SnpSIFT, VEP, UCSC gene symbols) -> variant filtering
```{r, echo=FALSE, message=FALSE}
panderOptions('table.split.table', Inf)
panderOptions('digits', 2)
pander(versions)
```
{to show what parameters were used, includes version number of curated files?}

***Ray's idea: document when a sample is archived, when it was zipped and sent to storage (name of sample, date, run, timestamp, genome build)***

**Date Analysis Performed:** `r format(Sys.time(), "%d %B, %Y")`

## Quality Metrics

Something from Proton/Torrent?

```{r VEPsummary, echo=FALSE, message=FALSE}
# general
generalStats <- scan('../vcf/DG515_NOTCH3_024_dbSNP_VEP.vcf_summary.txt', what = 'raw', sep = ',', skip = 12, nmax = 8)
generalStats <- gsub('\t', ' ', generalStats)
# variants
variantStats <- scan('../vcf/DG515_NOTCH3_024_dbSNP_VEP.vcf_summary.txt', what = 'raw', sep = ',', skip = 22, nmax = 5)
variantStats <- gsub('\t', ' ', variantStats)
# Consequences (most severe)
severeStats <- scan('../vcf/DG515_NOTCH3_024_dbSNP_VEP.vcf_summary.txt', what = 'raw', sep = ',', skip = 30, nmax = 20)
severeStats <- gsub('\t', ' ', severeStats)
# Coding consequences
codingStats <- scan('../vcf/DG515_NOTCH3_024_dbSNP_VEP.vcf_summary.txt', what = 'raw', sep = ',', skip = 81, nmax = 8)
codingStats <- gsub('\t', ' ', codingStats)
# SIFT summary
siftStats <- scan('../vcf/DG515_NOTCH3_024_dbSNP_VEP.vcf_summary.txt', what = 'raw', sep = ',', skip = 91, nmax = 4)
siftStats <- gsub('\t', ' ', siftStats)
# Polyphen summary
polyphenStats <- scan('../vcf/DG515_NOTCH3_024_dbSNP_VEP.vcf_summary.txt', what = 'raw', sep = ',', skip = 97, nmax = 4)
polyphenStats <- gsub('\t', ' ', polyphenStats)
```

### VEP summary statistics

#### General Statistics
```{r, echo=FALSE, results='asis'}
pandoc.list(generalStats[c(2,3,5:8)])
```

#### Variant Statistics
```{r, echo=FALSE, results='asis'}
pandoc.list(variantStats)
```

#### Consequences (most severe)
```{r, echo=FALSE, results='asis'}
pandoc.list(severeStats)
```

#### Coding consequences
```{r, echo=FALSE, results='asis'}
pandoc.list(codingStats)
```

#### SIFT summary
```{r, echo=FALSE, results='asis'}
pandoc.list(siftStats)
```

#### Polyphen summary
```{r, echo=FALSE, results='asis'}
pandoc.list(polyphenStats)
```

## Tier 1 Annotations – Disease Specific Genes 

{FHM, Ataxia, etc. For epilepsy and others, perhaps tiers could have specific names as there may be more tiers than the “standard” arrangement and some may be more likely to check for a specific patient }

### Tier 1.1 – Top 10 Most Damaging Mutations/Variants 

Any of the following:

  * Amino acid changing variant
    + no rs#
    + has rs#, but MAF <1%
  * No AA change
    + no rs#

{All sorted by the number of predictors showing “Damaging” or the equivalent}

```{r tier1 cleaning data, echo=FALSE, message=FALSE}
tier1 <- list.files('../results/Tier_1', pattern = '.csv', full.names = T)
tier1 <- tier1[grep(sampleID, tier1)]
tier1_table <- read.delim(tier1, head = T, as.is = T)
# tier1_table[tier1_table == ''] <- '.'
# tier1_table[is.na(tier1_table)] <- '.'
# gene
tier1_table$gene <- gsub('^.;', '', tier1_table$gene) 
tier1_table$gene <- sapply(sapply(strsplit(tier1_table$gene, ";"), unique), paste, collapse = ";")
# aa change
# tier1_table$amino_acid_substitution <- gsub(";$", "", tier1_table$amino_acid_substitution)
# tier1_table$amino_acid_substitution <- ifelse(tier1_table$amino_acid_substitution == ".", ".:.", tier1_table$amino_acid_substitution)
# SIFT
tier1_table$SIFT <- sapply(sapply(strsplit(tier1_table$SIFT, ","), unique), paste, collapse = ";")
# MutationTaster
tier1_table$MutationTaster <- sapply(sapply(strsplit(tier1_table$MutationTaster, ","), unique), paste, collapse = ";")
# Polyphen2
tier1_table$Polyphen2 <- sapply(sapply(strsplit(tier1_table$Polyphen2, ","), unique), paste, collapse = ";")
# 1KG allele freqs
tier1_table$CAF1[tier1_table$CAF1 == '.'] <- 0
tier1_table$CAF1 <- as.numeric(tier1_table$CAF1)
tier1_table$CAF2[tier1_table$CAF2 == '.'] <- 0
tier1_table$CAF2 <- as.numeric(tier1_table$CAF2)
#
tier1_clean <- data.frame(location = paste(tier1_table$chromosome, tier1_table$position, sep = ':'),
                      genotype = ifelse(tier1_table$genotype == '0/1', paste(tier1_table$reference_allele, tier1_table$alternate_allele, sep = '/'), ifelse(tier1_table$genotype == '1/1', paste(tier1_table$alternate_allele, tier1_table$alternate_allele, sep = '/'), tier1_table$genotype)),
                      ref = tier1_table$reference_allele, 
                      alt = tier1_table$alternate_allele,
                      dbSNP = tier1_table$id,
                      gene = tier1_table$gene,
                      transcript = strsplit(tier1_table$transcript, split = ';') %>% lapply(., `[[`, 1) %>% unlist(.),
                      coding = unlist(lapply(strsplit(tier1_table$coding, split = ';'), `[[`, 1)),
                      AAchange = unlist(lapply(strsplit(tier1_table$amino_acid_substitution, split = ';'), `[[`, 1)),
                      MutationTaster = tier1_table$MutationTaster,
                      SIFT = tier1_table$SIFT,
                      Polyphen2 = tier1_table$Polyphen2,
                      coverage = paste0(tier1_table$depth_coverage, ' ', tier1_table$reference_allele, '(', tier1_table$Ref_coverage, 
                                        ')', ' ', tier1_table$alternate_allele, '(', tier1_table$Alt_coverage, ')'),
                      ref_freq = tier1_table$CAF1,
                      alt_freq = tier1_table$CAF2)
# write clean table out to results dir
fileDate <- format(Sys.time(), "%a_%b_%d_%Y")
outFile <- paste0('../results/Tier_1/clean/', sampleID, '_Tier1_results_clean_', fileDate, '.csv')
write.csv(tier1_clean, outFile, row.names = F)
```

```{r table_1.1, echo=FALSE, message=FALSE}
tier1_mostdamg <- tier1_clean[tier1_clean$MutationTaster == "D",]
rownames(tier1_mostdamg) <- c(1:nrow(tier1_mostdamg))
# not to split tables
panderOptions('table.split.table', Inf)
panderOptions('digits', 2)
# panderOptions('table.style', 'simple')
# create table
emphasize.cols(6) # italics for gene names
# set.alignment(c('right', 'center'), row.names = 'right')
pander(tier1_mostdamg, caption = "Top 10 Most Damaging Mutations/Variants")
```
  
**Note:** *MutationTaster prediction*: **A** (disease causing automatic), **D** (disease causing), **N** (polymorphism) or **P** (polymorphism automatic). *SIFT prediction*: smaller than 0.05 is predicted as **D**(amaging), otherwise it is predicted as **T**(olerated). *Polyphen2 prediction*: **D** (probably damaging [0.957,1]), **P** (possibly damaging [0.453,0.956]) and **B** (benign [0,0.452]). Multiple predictions separated by ";".

### Tier 1.2 – Mutations 

```{r table_1.2 prep, echo=FALSE, message=FALSE, warning=FALSE}
tier1_mutations <- tier1_clean[tier1_clean$dbSNP == '.' & tier1_clean$AAchange != '.',]
if (nrow(tier1_mutations) != 0) rownames(tier1_mutations) <- c(1:nrow(tier1_mutations))
```
There are **`r nrow(tier1_mutations)`** variants that are amino acid changing and don't have an rs number.
```{r table_1.2, echo=FALSE, message=FALSE, warning=FALSE}
#
emphasize.cols(6) # italics for gene names
if (nrow(tier1_mutations) != 0) pander(tier1_mutations, caption = "Amino acid changing variant, no rs#")
```

### Tier 1.3 – Variants 

```{r table_1.3, echo=FALSE, message=FALSE, warning=FALSE}
tier1_3a <- tier1_clean[tier1_clean$dbSNP == '.' & tier1_clean$AAchange == ".",]
tier1_3b <- tier1_clean[tier1_clean$dbSNP != '.' & tier1_clean$AAchange != "." & tier1_clean$ref_freq < 0.01 | 
                          tier1_clean$dbSNP != '.' & tier1_clean$AAchange != "." &tier1_clean$alt_freq < 0.01,]
tier1_3 <- rbind(tier1_3a, tier1_3b)
```

There are **`r nrow(tier1_3a)`** variants that are not amino acid changing and have no associated rs number.  
There are **`r nrow(tier1_3b)`** variants that are amino acid changing, have a rs number and MAF < 1%.  

```{r table_1.3 out, echo=FALSE, message=FALSE, warning=FALSE}
rownames(tier1_3) <- c(1:nrow(tier1_3))
emphasize.cols(6) # italics for gene names
pander(tier1_3, caption = "Not amino acid changing and has no rs#; amino acid changing, has rs#, but MAF <1%")
```

### Tier 1.4 – Rare SNPs 

```{r table_1.4 out, echo=FALSE, message=FALSE, warning=FALSE}
tier1_4 <- tier1_clean[tier1_clean$dbSNP != '.' & (tier1_clean$ref_freq > 0.01 & tier1_clean$ref_freq < 0.1) | 
                        tier1_clean$dbSNP != '.' & (tier1_clean$alt_freq > 0.01 & tier1_clean$alt_freq < 0.1),]
rownames(tier1_4) <- c(1:nrow(tier1_4))
emphasize.cols(6) # italics for gene names
pander(tier1_4[c(1:10),], caption = "Any variant with rs number and MAF >1% <10%")
```

There are **`r nrow(tier1_4)`** variants classified as rare SNPs (top 10 listed above).  
**NB:** these are only the first 10 of the filtered list...

\newpage

## Tier 2 Annotations – Pathway Specific Genes 

{CNS, Muscle, Cardiac, etc. Drawn from KEGG or similar database?}

### Tier 2.1 – Top 10 Most Damaging Mutations/Variants 

Any of the following:

  * Amino acid changing variant
    + no rs#
    + has rs#, but MAF <1%
  * No AA change
    + no rs#

{All sorted by the number of predictors showing “Damaging” or the equivalent}

```{r tier2 cleaning data, echo=FALSE, message=FALSE}
tier2 <- list.files('../results/Tier_2', pattern = '.csv', full.names = T)
tier2 <- tier2[grep(sampleID, tier2)]
tier2_table <- read.delim(tier2, head = T, as.is = T)
# tier2_table[tier2_table == ''] <- '.'
# tier2_table[is.na(tier2_table)] <- '.'
# gene
tier2_table$gene <- gsub('^.;', '', tier2_table$gene) 
tier2_table$gene <- sapply(sapply(strsplit(tier2_table$gene, ";"), unique), paste, collapse = ";")
# aa change
# tier2_table$amino_acid_substitution <- gsub(";$", "", tier2_table$amino_acid_substitution)
# tier2_table$amino_acid_substitution <- ifelse(tier2_table$amino_acid_substitution == ".", ".:.", tier2_table$amino_acid_substitution)
# SIFT
tier2_table$SIFT <- sapply(sapply(strsplit(tier2_table$SIFT, ","), unique), paste, collapse = ";")
# MutationTaster
tier2_table$MutationTaster <- sapply(sapply(strsplit(tier2_table$MutationTaster, ","), unique), paste, collapse = ";")
# Polyphen2
tier2_table$Polyphen2 <- sapply(sapply(strsplit(tier2_table$Polyphen2, ","), unique), paste, collapse = ";")
# 1KG allele freqs
tier2_table$CAF1[tier2_table$CAF1 == '.'] <- 0
tier2_table$CAF1 <- as.numeric(tier2_table$CAF1)
tier2_table$CAF2[tier2_table$CAF2 == '.'] <- 0
tier2_table$CAF2 <- as.numeric(tier2_table$CAF2)
#
tier2_clean <- data.frame(location = paste(tier2_table$chromosome, tier2_table$position, sep = ':'),
                      genotype = ifelse(tier2_table$genotype == '0/1', paste(tier2_table$reference_allele, tier2_table$alternate_allele, sep = '/'), ifelse(tier2_table$genotype == '1/1', paste(tier2_table$alternate_allele, tier2_table$alternate_allele, sep = '/'), tier2_table$genotype)),
                      ref = tier2_table$reference_allele, 
                      alt = tier2_table$alternate_allele,
                      dbSNP = tier2_table$id,
                      gene = tier2_table$gene,
                      transcript = strsplit(tier2_table$transcript, split = ';') %>% lapply(., `[[`, 1) %>% unlist(.),
                      coding = unlist(lapply(strsplit(tier2_table$coding, split = ';'), `[[`, 1)),
                      AAchange = unlist(lapply(strsplit(tier2_table$amino_acid_substitution, split = ';'), `[[`, 1)),
                      MutationTaster = tier2_table$MutationTaster,
                      SIFT = tier2_table$SIFT,
                      Polyphen2 = tier2_table$Polyphen2,
                      coverage = paste0(tier2_table$depth_coverage, ' ', tier2_table$reference_allele, '(', tier2_table$Ref_coverage, 
                                        ')', ' ', tier2_table$alternate_allele, '(', tier2_table$Alt_coverage, ')'),
                      ref_freq = tier2_table$CAF1,
                      alt_freq = tier2_table$CAF2)
# write clean table out to results dir
fileDate <- format(Sys.time(), "%a_%b_%d_%Y")
outFile <- paste0('../results/Tier_2/clean/', sampleID, '_Tier2_results_clean_', fileDate, '.csv')
write.csv(tier2_clean, outFile, row.names = F)
```

```{r table_2.1, echo=FALSE, message=FALSE}
tier2_mostdamg <- tier2_clean[tier2_clean$MutationTaster == "D",]
tier2_mostdamg <- tier2_clean[tier2_clean$MutationTaster == "D" & tier2_clean$SIFT == "D" & tier2_clean$Polyphen2 == "D" &
                                tier2_clean$dbSNP == "." & tier2_clean$AAchange != '.',]
# tier2_mostdamg <- tier2_clean[tier2_clean$dbSNP != '.' & tier2_clean$AAchange != '.' & tier2_clean$ref_freq <= 0.01,]
# tier2_mostdamg <- tier2_clean[tier2_clean$dbSNP == '.' & tier2_clean$AAchange == '.',]
# tier2_mostdamg <- tier2_mostdamg[tier2_mostdamg$dbSNP == '.' & tier2_mostdamg$AAchange != '.' | 
#                                    tier2_mostdamg$dbSNP != '.' & tier2_mostdamg$AAchange != '.' & tier2_mostdamg$ref_freq <= 0.01, ]
rownames(tier2_mostdamg) <- c(1:nrow(tier2_mostdamg))
# set.alignment(c('right', 'center'), row.names = 'right')
emphasize.cols(6) # italics for gene names
pander(tier2_mostdamg, caption = "Top 10 Most Damaging Mutations/Variants")
```

### Tier 2.2 – Mutations 

There are **`r nrow(tier2_clean[tier2_clean$dbSNP == '.' & tier2_clean$AAchange != '.',])`** variants that are amino acid changing without associated rs numbers. 

```{r table_2.2, echo=FALSE, message=FALSE}
tier2_mutations <- tier2_clean[tier2_clean$dbSNP == '.' & tier2_clean$AAchange != '.',]
rownames(tier2_mutations) <- c(1:nrow(tier2_mutations))
#
emphasize.cols(6) # italics for gene names
pander(tier2_mutations[c(1:10),], caption = "Amino acid changing variant, no rs#")
```

### Tier 2.3 – Variants 

```{r table_2.3, echo=FALSE, message=FALSE, warning=FALSE}
tier2_3a <- tier2_clean[tier2_clean$dbSNP == '.' & tier2_clean$AAchange == ".",]
tier2_3b <- tier2_clean[tier2_clean$dbSNP != '.' & tier2_clean$AAchange != "." & tier2_clean$ref_freq < 0.01 | 
                          tier2_clean$dbSNP != '.' & tier2_clean$AAchange != "." &tier2_clean$alt_freq < 0.01,]
tier2_3 <- rbind(tier2_3a, tier2_3b)
```

There are **`r nrow(tier2_3a)`** variants that are not amino acid changing and have no associated rs number.  
There are **`r nrow(tier2_3b)`** variants that are amino acid changing, have a rs number and MAF < 1%.  

```{r table_2.3 out, echo=FALSE, message=FALSE, warning=FALSE}
rownames(tier2_3) <- c(1:nrow(tier2_3))
emphasize.cols(6) # italics for gene names
pander(tier2_3[c(1:10),], caption = "Not amino acid changing and has no rs#; amino acid changing, has rs#, but MAF <1%")
```

### Tier 2.4 – Rare SNPs 

```{r table_2.4 out, echo=FALSE, message=FALSE, warning=FALSE}
tier2_4 <- tier2_clean[tier2_clean$dbSNP != '.' & (tier2_clean$ref_freq > 0.01 & tier2_clean$ref_freq < 0.1) | 
                        tier2_clean$dbSNP != '.' & (tier2_clean$alt_freq > 0.01 & tier2_clean$alt_freq < 0.1),]
rownames(tier2_4) <- c(1:nrow(tier2_4))
emphasize.cols(6) # italics for gene names
pander(tier2_4[c(1:10),], caption = "Any variant with rs number and MAF >1% <10%")
```

There are **`r nrow(tier2_4)`** variants classified as rare SNPs (top 10 listed above).
**NB:** these are only the first 10 of the filtered list...

\newpage

##Tier 3 Annotations – All Other Genes 

{Anything not included in the previous sets. I would suggest perhaps moving the Rare SNPs for this tier down to Tier 4, as there will be too many to work with. That may also need to be done for the Epilepsy or other panels where there’s more than 30 or so targeted genes in a tier}.

### Tier 3.1 – Top 10 Most Damaging Mutations/Variants 

Any of the following:

  * Amino acid changing variant
    + no rs#
    + has rs#, but MAF <1%
  * No AA change
    + no rs#

{All sorted by the number of predictors showing “Damaging” or the equivalent}

```{r tier3 cleaning data, echo=FALSE, message=FALSE}
tier3 <- list.files('../results/Tier_3', pattern = '.csv', full.names = T)
tier3 <- tier3[grep(sampleID, tier3)]
tier3_table <- read.delim(tier3, head = T, as.is = T)
# tier3_table[tier3_table == ''] <- '.'
# tier3_table[is.na(tier3_table)] <- '.'
# gene
tier3_table$gene <- gsub('^.;', '', tier3_table$gene) 
tier3_table$gene <- sapply(sapply(strsplit(tier3_table$gene, ";"), unique), paste, collapse = ";")
# aa change
# tier3_table$amino_acid_substitution <- gsub(";$", "", tier3_table$amino_acid_substitution)
# tier3_table$amino_acid_substitution <- ifelse(tier3_table$amino_acid_substitution == ".", ".:.", tier3_table$amino_acid_substitution)
# SIFT
tier3_table$SIFT <- sapply(sapply(strsplit(tier3_table$SIFT, ","), unique), paste, collapse = ";")
# MutationTaster
tier3_table$MutationTaster <- sapply(sapply(strsplit(tier3_table$MutationTaster, ","), unique), paste, collapse = ";")
# Polyphen2
tier3_table$Polyphen2 <- sapply(sapply(strsplit(tier3_table$Polyphen2, ","), unique), paste, collapse = ";")
# 1KG allele freqs
tier3_table$CAF1[tier3_table$CAF1 == '.'] <- 0
tier3_table$CAF1 <- as.numeric(tier3_table$CAF1)
tier3_table$CAF2[tier3_table$CAF2 == '.'] <- 0
tier3_table$CAF2 <- as.numeric(tier3_table$CAF2)
#
tier3_clean <- data.frame(location = paste(tier3_table$chromosome, tier3_table$position, sep = ':'),
                      genotype = ifelse(tier3_table$genotype == '0/1', paste(tier3_table$reference_allele, tier3_table$alternate_allele, sep = '/'), ifelse(tier3_table$genotype == '1/1', paste(tier3_table$alternate_allele, tier3_table$alternate_allele, sep = '/'), tier3_table$genotype)),
                      ref = tier3_table$reference_allele, 
                      alt = tier3_table$alternate_allele,
                      dbSNP = tier3_table$id,
                      gene = tier3_table$gene,
                      transcript = strsplit(tier3_table$transcript, split = ';') %>% lapply(., `[[`, 1) %>% unlist(.),
                      coding = unlist(lapply(strsplit(tier3_table$coding, split = ';'), `[[`, 1)),
                      AAchange = unlist(lapply(strsplit(tier3_table$amino_acid_substitution, split = ';'), `[[`, 1)),
                      MutationTaster = tier3_table$MutationTaster,
                      SIFT = tier3_table$SIFT,
                      Polyphen2 = tier3_table$Polyphen2,
                      coverage = paste0(tier3_table$depth_coverage, ' ', tier3_table$reference_allele, '(', tier3_table$Ref_coverage, 
                                        ')', ' ', tier3_table$alternate_allele, '(', tier3_table$Alt_coverage, ')'),
                      ref_freq = tier3_table$CAF1,
                      alt_freq = tier3_table$CAF2)
# write clean table out to results dir
fileDate <- format(Sys.time(), "%a_%b_%d_%Y")
outFile <- paste0('../results/Tier_3/clean/', sampleID, '_Tier3_results_clean_', fileDate, '.csv')
write.csv(tier3_clean, outFile, row.names = F)
```

```{r table_3.1, echo=FALSE, message=FALSE}
tier3_mostdamg <- tier3_clean[tier3_clean$MutationTaster == "D",]
# tier3_mostdamg <- tier3_clean[tier3_clean$MutationTaster == "D" & tier3_clean$SIFT == "D" & tier3_clean$Polyphen2 == "D" &
#                                 tier3_clean$dbSNP == "." & tier3_clean$AAchange != '.',]
# tier3_mostdamg <- tier3_clean[tier3_clean$dbSNP != '.' & tier3_clean$AAchange != '.' & tier3_clean$ref_freq <= 0.01,]
# tier3_mostdamg <- tier3_clean[tier3_clean$dbSNP == '.' & tier3_clean$AAchange == '.',]
# tier3_mostdamg <- tier3_mostdamg[tier3_mostdamg$dbSNP == '.' & tier3_mostdamg$AAchange != '.' | 
#                                    tier3_mostdamg$dbSNP != '.' & tier3_mostdamg$AAchange != '.' & tier3_mostdamg$ref_freq <= 0.01, ]
rownames(tier3_mostdamg) <- c(1:nrow(tier3_mostdamg))
# set.alignment(c('right', 'center'), row.names = 'right')
emphasize.cols(6) # italics for gene names
pander(tier3_mostdamg[c(1:10),], caption = "Top 10 Most Damaging Mutations/Variants")
```

### Tier 3.2 – Mutations 

{Amino acid changing variant, no rs#}

### Tier 3.3 – Variants 

{Not amino acid changing and has no rs#; amino acid changing, has rs#, but MAF <1%}

### Tier 3.4 – Rare SNPs 

{any variant with rs# and MAF >1%<10%}

\newpage

## Tier 4 Annotations – Polymorphisms 

{Polymorphisms for all Tier }.

### Tier 4.1 – Polymorphisms for Tier 1 Genes 

{any variant with rs# and MAF >10%}

### Tier 4.2 – Polymorphisms for Tier 2 Genes 

{any variant with rs# and MAF >10%}

### Tier 4.3 – Polymorphisms for Tier 3 Genes 

{any variant with rs# and MAF >10%}
